src/test/ui/layout/gc-layout.rs

```diff
@@ -0,0 +1,88 @@
+// run-pass
+
+#![feature(gc)]
+
+#![allow(dead_code)]
+#![allow(unused_attributes)]
+#![allow(unused_imports)]
+
+use std::gc::gc_layout;
+use std::mem::size_of;
+use std::mem::align_of;
+
+struct SmallAlign {
+    a: u8,
+    b: u16,
+    c: u8
+}
+
+struct ShuffledFields {
+    a: u8,
+    b: u64,
+    c: u8
+}
+
+struct ShuffledFieldsNoTrace {
+    a: u8,
+    #[no_trace]
+    b: u64,
+    c: u8
+}
+
+#[repr(C)]
+struct FixedFields {
+    a: u8,
+    b: u64,
+    c: u8
+}
+
+#[repr(C)]
+struct FixedFieldsNoTrace {
+    a: u8,
+    #[no_trace]
+    b: u64,
+    c: u8
+}
+
+struct MultiFields {
+    a: u8,
+    b: u16,
+    e: usize,
+    c: u32,
+    d: u8,
+    #[no_trace]
+    f: usize
+}
+
+struct BigAlign {
+    a: MultiFields,
+    b: usize,
+    c: usize,
+}
+
+struct BigAlignNoTrace {
+    b: usize,
+    c: usize,
+    #[no_trace]
+    a: MultiFields,
+}
```

Comment:
Maybe add some edge cases? The empty struct? A `#[repr(C)]` struct? Do enums work?

Reply:
No enums are not supported yet. Sure, I can add a couple of edge cases.

Reply:
Here is the ZST case: eb9cd6f706200a93072e904c1ba988df365ac0b6